File: demos/node/removeBg.mjs
import 'dotenv/config';

import { GetObjectCommand, PutObjectCommand } from '@aws-sdk/client-s3';
import { S3Client } from "@aws-sdk/client-s3";
import { getSignedUrl } from "@aws-sdk/s3-request-presigner";

const s3Client = new S3Client({ region: 'us-east-1' });
const bucket = 'psapitestrkc';

const CLIENT_ID = process.env.CLIENT_ID;
const CLIENT_SECRET = process.env.CLIENT_SECRET;

async function getSignedDownloadUrl(path) {
	let command = new GetObjectCommand({ Bucket: bucket, Key:path });
	return await getSignedUrl(s3Client, command, { expiresIn: 3600 });
}

async function getSignedUploadUrl(path) {
	let command = new PutObjectCommand({ Bucket: bucket, Key:path });
	return await getSignedUrl(s3Client, command, { expiresIn: 3600 });
}

async function getAccessToken(id,secret) {

	const params = new URLSearchParams();
	params.append('client_secret', secret);
	params.append('grant_type', 'client_credentials');
	params.append('scope', 'openid,AdobeID,read_organizations');

	let resp = await fetch(`https://ims-na1.adobelogin.com/ims/token/v2?client_id=${id}`, 
		{ 
			method: 'POST', 
			body: params
		}
	);

	let data = await resp.json();
	return data.access_token;

}

(async () => {

	let token = await getAccessToken(CLIENT_ID, CLIENT_SECRET);

	let inputURL = await getSignedDownloadUrl('input/dogs.jpg');
	let uploadURL = await getSignedUploadUrl('output/dogs_nobg.jpg');

	let data = {
		"input": {
			"href": inputURL,
			"storage": "external"
		},
		"options": {
			"optimize": "performance",
		},
		"output": {
			"href": uploadURL,
			"storage": "external",
			"overwrite": true,
			"color": {
				"space": "rgb"
			},
			"mask": {
				"format": "soft"
			}
		}
	};				

	let resp = await fetch('https://image.adobe.io/sensei/cutout', {
		headers: {
			'Authorization':`Bearer ${token}`,
			'x-api-key': CLIENT_ID,
			'Content-Type':'application/json'
		}, 
		method:'POST',
		body:JSON.stringify(data)
	});

	let job = await resp.json();
	console.log(job);

	/*
	 In theory, we can just assume everything worked.
	 That's never backfired on anyone, right?
	*/

})()

File: demos/python/removeBg.py
import os 
import requests 
import boto3
from botocore.exceptions import ClientError
from botocore.config import Config
import time 

CLIENT_ID = os.environ.get('CLIENT_ID')
CLIENT_SECRET = os.environ.get('CLIENT_SECRET')
BUCKET = 'psapitestrkc'
REGION = 'us-east-1'

def getAccessToken(id, secret):

	response = requests.post(f"https://ims-na1.adobelogin.com/ims/token/v2?client_id={id}&client_secret={secret}&grant_type=client_credentials&scope=openid,AdobeID")
	return response.json()


def create_removebg_job(id, token, inputURL, uploadURL):
           
	data = {
		"input": {
			"href": inputURL,
			"storage": "external"
		},
		"options": {
			"optimize": "performance",
		},
		"output": {
			"href": uploadURL,
			"storage": "external",
			"overwrite": True,
			"color": {
				"space": "rgb"
			},
			"mask": {
				"format": "soft"
			}
		}
	}			

	response = requests.post(f"https://image.adobe.io/sensei/cutout", headers = {"Authorization": f"Bearer {token}", "x-api-key": id }, json=data)
	return response.json()


# Credit: https://github.com/AdobeDocs/cis-photoshop-api-docs/blob/main/sample-code/storage-app/aws-s3/example.py
def get_presigned_url(bucket, key, operation, expires_in, region):
    s3 = boto3.client('s3', config=Config(signature_version='s3v4'), region_name=region)
    url = s3.generate_presigned_url(operation, Params={'Bucket': bucket, 'Key': key}, ExpiresIn=expires_in)
    return url        

token = getAccessToken(CLIENT_ID, CLIENT_SECRET)['access_token']

inputURL = get_presigned_url(BUCKET, 'input/dogs.jpg', 'get_object', 3600, REGION)
uploadURL = get_presigned_url(BUCKET, 'output/dogs_nobg_python.jpg', 'put_object', 3600, REGION)

job = create_removebg_job(CLIENT_ID, token, inputURL, uploadURL)
print(job)